pkg/broker/instance_operations.go

```diff
@@ -37,21 +40,78 @@ func (b Broker) Provision(ctx context.Context, instanceID string, details broker
 	if err != nil {
-		b.logger.Errorw("Failed to create Atlas cluster", "error", err, "cluster", cluster)
-		err = atlasToAPIError(err)
-		return
+		// Create a new Atlas cluster from the generated definition
+		resultingCluster, err = b.atlas.CreateCluster(*cluster)
+		if err != nil {
+			b.logger.Errorw("Failed to create Atlas cluster", "error", err, "cluster", cluster)
+			err = atlasToAPIError(err)
+			return
+		}
+
+		b.logger.Infow("Successfully started Atlas creation process", "instance_id", instanceID, "cluster", resultingCluster)
+
+		return brokerapi.ProvisionedServiceSpec{
+			IsAsync:       true,
+			OperationData: OperationProvision,
+			DashboardURL:  b.atlas.GetDashboardURL(resultingCluster.Name),
+		}, nil
 	}
 
-	b.logger.Infow("Successfully started Atlas creation process", "instance_id", instanceID, "cluster", resultingCluster)
+	err = CompareAndReturnAppropiateResponseCode(resultingCluster, cluster)
+	return
+}
 
-	return brokerapi.ProvisionedServiceSpec{
-		IsAsync:       true,
-		OperationData: OperationProvision,
-		DashboardURL:  b.atlas.GetDashboardURL(resultingCluster.Name),
-	}, nil
+// CompareAndReturnAppropiateResponseCode converts structs to maps and afterwards returns the appropiate response code
+func CompareAndReturnAppropiateResponseCode(resultingCluster *atlas.Cluster, cluster *atlas.Cluster) error {
+	//Convert structs to maps
+	var remoteClusterInterface map[string]interface{}
+	var localClusterInterface map[string]interface{}
+	inrec, _ := json.Marshal(resultingCluster)
+	json.Unmarshal(inrec, &remoteClusterInterface)
+
+	inrec, _ = json.Marshal(cluster)
+	json.Unmarshal(inrec, &localClusterInterface)
+
+	if compareHelper(remoteClusterInterface, localClusterInterface) {
+		return nil
+	}
+	return apiresponses.NewFailureResponse(errors.New("There IDs are equal but differ in their attributes"), http.StatusConflict, "")
+}
+
+func compareHelper(remoteClusterInterface map[string]interface{}, localClusterInterface map[string]interface{}) bool {
+	if reflect.ValueOf(localClusterInterface).Kind() == reflect.Map {
+		for k, v := range localClusterInterface {
+			if reflect.ValueOf(v).Kind() == reflect.Map && len(v.(map[string]interface{})) != 0 {
+				equal := compareHelper(remoteClusterInterface[k].(map[string]interface{}), v.(map[string]interface{}))
+				if !equal {
+					return false
+				}
+				continue
+			} else if reflect.ValueOf(v).Kind() == reflect.Slice && len(v.([]interface{})) != 0 {
+				for index, document := range v.([]interface{}) { // assuming every item in slice is a document
+					equal := compareHelper(remoteClusterInterface[k].([]interface{})[index].(map[string]interface{}), document.(map[string]interface{}))
+					if !equal {
+						return false
+					}
+				}
+			} else if reflect.ValueOf(v).Kind() != reflect.Map && reflect.ValueOf(v).Kind() != reflect.Slice { // others are not maps nor slices, but rather other types
+				if val, ok := remoteClusterInterface[k]; ok {
```

Comment:
These could be two different types completely, making the `v != val` comparison to panic